sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -374,7 +375,10 @@ case class AlterTableChangeColumnCommand(
   private def findColumnByName(
       schema: StructType, name: String, resolver: Resolver): StructField = {
     schema.fields.collectFirst {
-      case field if resolver(field.name, name) => field
+      case field if resolver(field.name, name) =>
+        CharVarcharUtils.getRawType(field.metadata)
```

Comment:
It looks weird to restore char/varchar in `findColumnByName`. Can we do it in the caller side? The char/varchar metaddata can appear in the top-level struct field anyway.

Reply:
right, we can getTableMetadata with the raw  scheam definition directly here